beam-migrate/Database/Beam/Migrate/Generics/Types.hs

```diff
@@ -27,3 +29,33 @@ instance ( Selector f, IsCheckedDatabaseEntity be x
 
   defaultMigratableDbSettings' _ = M1 (K1 (CheckedDatabaseEntity (checkedDbEntityAuto name) []))
     where name = T.pack (selName (undefined :: S1 f (Rec0 (CheckedDatabaseEntity be db x)) ()))
+
+instance ( Selector f
+         , Generic (CheckedDatabaseSettings be innerDb)
+         , GAutoMigratableDb be (Rep (CheckedDatabaseSettings be innerDb))
+         , Database be innerDb
+         ) =>
+  GAutoMigratableDb be (S1 f (Rec0 (innerDb (CheckedDatabaseEntity be outerDb)))) where
+
+  defaultMigratableDbSettings' _ = M1 (K1 (runIdentity $ zipTables (Proxy :: Proxy be) (\_ -> pure . changeCheckedDatabaseEntityDb @innerDb @outerDb @be) settings settings :: innerDb (CheckedDatabaseEntity be outerDb))) --TODO: Rename inner tables based on field name
```

Comment:
At the very least this `TODO` needs to be removed. I'm not super convinced one way or another that inner tables should be prefixed with the container field name, but whatever way we go with we should stick with as users depend on it for migrations to work.

Reply:
This has been resolved